ddtrace/contrib/internal/openai/utils.py

```diff
@@ -249,26 +275,15 @@ def _construct_message_from_streamed_chunks(streamed_chunks: List[Any]) -> Dict[
             continue
         function_call = getattr(chunk.delta, "function_call", None)
         if function_call:
-            if idx is None:
-                formatted_content += "\n\n[function: {}]\n\n".format(getattr(function_call, "name", ""))
-                idx = chunk.index
-            function_args = getattr(function_call, "arguments", "")
-            message["content"] += "{}".format(function_args)
-            formatted_content += "{}".format(function_args)
-        tool_calls = getattr(chunk.delta, "tool_calls", [])
+            _construct_tool_call_from_streamed_chunk(message["tool_calls"], function_call_chunk=function_call)
+        tool_calls = getattr(chunk.delta, "tool_calls", None)
         if not tool_calls:
             continue
         for tool_call in tool_calls:
-            if tool_call.index != idx:
-                formatted_content += "\n\n[tool: {}]\n\n".format(getattr(tool_call.function, "name", ""))
-                idx = tool_call.index
-            function_args = getattr(tool_call.function, "arguments", "")
-            message["content"] += "{}".format(function_args)
-            formatted_content += "{}".format(function_args)
-
+            _construct_tool_call_from_streamed_chunk(message["tool_calls"], tool_call_chunk=tool_call)
+    if message["tool_calls"]:
+        message["tool_calls"].sort(key=lambda x: x.get("index", 0))
     message["content"] = message["content"].strip()
```

Comment:
Should we do a check here to pop the `tool_calls` key if it is an empty list? Just to avoid a situation where it is accidentally displayed in the FE
```
if message.get("tool_calls") is not None and len(message.get("tool_calls")) == 0:
      message.pop("tool_calls")
```